File: Wayne/WayneCompiler/src/Dfir_representation/IFeedbackInputNode.java
package Dfir_representation;

import java.util.Vector;


public class IFeedbackInputNode extends JNode{
	private int delay;
	private Boolean IsFeedbackDeclared;
	private OutputNode outputNode;
	   public IFeedbackInputNode(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals, int delay, OutputNode outputNode){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   this.delay = delay;
		   this.outputNode = outputNode;
		   JNodeType = "IFeedbackInputNode";
		   IsFeedbackDeclared = false;
	   }
	  
	   public void print(){
		   System.out.println("IFeedbackInputNode begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
		   System.out.println("delay: " + this.delay);
		   this.outputNode.print();
		   for(int i=0; i<inputTerminals.size(); i++){
				   inputTerminals.get(i).print();
		   }
		  for(int i=0; i<outputTerminals.size(); i++){
				   outputTerminals.get(i).print();
		   }   
		  System.out.println("IFeedbackInputNode end\n");
	   }
	   
	   public IFeedbackOutputNode getIFeedbackOutputNode(){
	    	return (IFeedbackOutputNode)JNode.getIdToJNode().get(outputNode.getNodeId());
	    }
	   
    public String getFeedbackName(){
    	return "Feedback" + Integer.toString(this.getNodeId());
    }
	public void setIsFeedbackDeclared(Boolean isFeedbackDeclared) {
		IsFeedbackDeclared = isFeedbackDeclared;
	}

	public Boolean getIsFeedbackDeclared(){
		return this.IsFeedbackDeclared;
	}
	public int getdelay() {
		return delay;
	}

	public void setdelay(int delay) {
		this.delay = delay;
	}

	public OutputNode getOutputNode() {
		return outputNode;
	}

	public void setOutputNode(OutputNode outputNode) {
		this.outputNode = outputNode;
	}
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/IFeedbackInputNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DFIR_Compiler
{
    public class IFeedbackInputNode:INode
    {
        private int delay;
        private bool isFeedbackDeclared;
        private OutputNode outputNode;

        public IFeedbackInputNode(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals, int Delay, OutputNode _outputNode) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            delay = Delay;
            outputNode = _outputNode;
            isFeedbackDeclared = false;
            INodeType = "IFeedbackInputNode";
        }


        public virtual int GetDalay()
        {
            return delay;
        }
        public virtual bool GetIsFeedbackDeclared()
        {
            return isFeedbackDeclared;
        }
        public virtual OutputNode GetOutputNode()
        {
            return outputNode;
        }

        public virtual IFeedbackOutputNode GetIFeedbackOutputNode()
        {
            return (IFeedbackOutputNode)GetIdToINode()[outputNode.GetNodeId()];
        }

        public virtual string GetFeedbackName()
        {
            return "Feedback" + Convert.ToString(GetNodeId());
        }


        public virtual void SetDalay(int Delay)
        {
            delay = Delay;
        }
        public virtual void SetIsFeedbackDeclared(bool isFeedbackDeclared)
        {
            this.isFeedbackDeclared = isFeedbackDeclared;
        }
        public virtual void SetOutputNode(OutputNode outputNode)
        {
            this.outputNode = outputNode;
        }


        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("IFeedbackInputNode:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
            Console.WriteLine("Dalay: " + delay);
            outputNode.Print();
            for (int i = 0; i < inputTerminals.Count(); i++)
            {
                inputTerminals[i].Print();
            }
            for (int i = 0; i < outputTerminals.Count(); i++)
            {
                outputTerminals[i].Print();
            }
        }
    }
}
